node/pkg/telemetry/telemetry.go

```diff
@@ -14,16 +17,53 @@ import (
 	"google.golang.org/api/option"
 )
 
+const TELEMETRY_LOG_LEVEL = zap.InfoLevel
+
 type Telemetry struct {
-	encoder            *guardianTelemetryEncoder
-	serviceAccountJSON []byte
+	encoder *guardianTelemetryEncoder
+}
+
+type ExternalLogger interface {
+	log(time time.Time, message []byte, level zapcore.Level)
+	flush() error
+}
+
+type ExternalLoggerGoogleCloud struct {
+	*logging.Logger
+	labels map[string]string // labels to add to each cloud log
+}
+
+func (logger *ExternalLoggerGoogleCloud) log(time time.Time, message []byte, level zapcore.Level) {
+	logger.Log(logging.Entry{
+		Timestamp: time,
+		Payload:   message,
+		Severity:  logLevelSeverity[level],
+		Labels:    logger.labels,
+	})
+}
+
+func (logger *ExternalLoggerGoogleCloud) flush() error {
+	return logger.Flush()
+}
+
+// Mock external logger for integration tests
```

Comment:
Not entirely comfortable with calling this "Mock", since it will actually be used in production if telemetry is disabled. At least update the comment to indicate that it is not only used for tests. Should there be a separate "null logger" used in prod that doesn't include the event counter?

Reply:
renamed it to `ExternalLoggerNil`